sdk/storage/Azure.Storage.Files.DataLake/src/Models/PathPermissions.cs

```diff
@@ -0,0 +1,238 @@
+            ExtendedInfoInAcl = extendedInfoInAcl;
+        }
+
+        /// <summary>
+        /// Parses a string in octal format to PathPermissions.
+        /// </summary>
+        /// <param name="octal">Octal string to parse.</param>
+        /// <returns><see cref="PathPermissions"/>.</returns>
+        public static PathPermissions ParseOctal(string octal)
+        {
+            if (octal == null)
+            {
+                throw Errors.ArgumentNull(nameof(octal));
+            }
+
+            if (octal.Length != 4)
+            {
+                throw Errors.InvalidArgument(nameof(octal));
+            }
+
+            var pathPermissions = new PathPermissions();
+
+            if (octal[0] == '0')
+            {
+                pathPermissions.StickyBit = false;
+            }
+            else
+            {
+                pathPermissions.StickyBit = true;
+            }
+
+            pathPermissions.Owner = RolePermissionsExtensions.ParseOctal(octal[1]);
+            pathPermissions.Group = RolePermissionsExtensions.ParseOctal(octal[2]);
+            pathPermissions.Owner = RolePermissionsExtensions.ParseOctal(octal[3]);
+
+            return pathPermissions;
+        }
+
+        /// <summary>
+        /// Parses a symbolic string to PathPermissions.
+        /// </summary>
+        /// <param name="str">String to parse</param>
+        /// <returns><see cref="PathPermissions"/></returns>
+        public static PathPermissions ParseSymbolic(string str)
+        {
+            if (str == null)
+            {
+                throw Errors.ArgumentNull(nameof(str));
+            }
+
+            if (str.Length != 9 && str.Length != 10)
+            {
+                throw Errors.InvalidArgument(nameof(str));
+            }
+
+            var pathPermissions = new PathPermissions();
+
+            // Set sticky bit
+            if (str.ToLower(CultureInfo.InvariantCulture)[str.Length - 1] == 't')
+            {
+                pathPermissions.StickyBit = true;
+            }
+            else
+            {
+                pathPermissions.StickyBit = false;
+            }
+
+            // Set extended info in ACL
+            if (str.Length == 10)
+            {
+                if (str[9] == '+')
+                {
+                    pathPermissions.ExtendedInfoInAcl = true;
+                }
+                else
+                {
+                    throw Errors.InvalidFormat(nameof(str));
+                }
+            }
+            else
+            {
+                pathPermissions.ExtendedInfoInAcl = false;
+            }
+
+            pathPermissions.Owner = RolePermissionsExtensions.ParseSymbolic(str.Substring(0, 3), allowStickyBit: false);
+            pathPermissions.Group = RolePermissionsExtensions.ParseSymbolic(str.Substring(3, 3), allowStickyBit: false);
+            pathPermissions.Other = RolePermissionsExtensions.ParseSymbolic(str.Substring(6, 3), allowStickyBit: true);
```

Comment:
pass pathPermissions.StickyBit instead of true

Reply:
It's passing the new unit tests as-is.